### PR TITLE
run ansible-pull from the right dir, with HOME set

### PR DESCRIPTION
**Pull request by djmitche**
This should fix [TRAC-3132](http://trac.buildbot.net/ticket/3132).  It runs under `env -i` anyway!

### CODE REVIEW

**Comment by djmitche** [comment]
To deploy, I'll need to run `ansible-pull` manually on every host and jail.

File: roles/base/templates/run-ansible-pull.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+cd /home/{{ service_account}}/repo
+ansible-pull \
+    -C master \
+    -d /home/{{ service_account }}/repo/ \
+    -m git \
+    -U {{ ansible_git_repository }} \
+    -o \
+    -s 3600 \
+    -i localhost \
+    --vault-password-file=/home/{{ service_account }}/.vault-password
```
Comment (seankelly) [inline roles/base/templates/run-ansible-pull.sh.j2]: Missing a trailing backslash.

Comment (djmitche) [inline roles/base/templates/run-ansible-pull.sh.j2]: thanks (forgot to push that fix)

File: localhost [hunk 2]
```diff
@@ -1 +1 @@
-localhost
+localhost ansible_python_interpreter=/usr/local/bin/python2.7
```
Comment (sa2ajj) [inline localhost]: Should we make the value in `group_vars/all` an absolute path instead?

File: roles/base/templates/run-ansible-pull.sh.j2 [hunk 3]
```diff
@@ -0,0 +1,18 @@
+#! /bin/sh
+
+# set some environment variables that ansible-pull needs
+export HOME=/home/{{ service_account }}
+export PATH=/sbin:/bin:/usr/sbin:/usr/bin:/usr/games:/usr/local/sbin:/usr/local/bin:/home/{{ service_account }}/bin
```
Comment (sa2ajj) [inline roles/base/templates/run-ansible-pull.sh.j2]: do we have anything in service account's `bin/`?

File: localhost (hunk 2 above)
Comment (djmitche) [inline localhost]: That might help.  I'm not sure if that's why this wasn't working, or if ansible-pull's git checkout step doesn't use `group_vars/all`

File: roles/base/templates/run-ansible-pull.sh.j2 (hunk 3 above)
Comment (djmitche) [inline roles/base/templates/run-ansible-pull.sh.j2]: I don't think so, but it was part of the PATH on login, so I put it here.

File: localhost (hunk 2 above)
Comment (sa2ajj) [inline localhost]: Do you have any logs when it did not work?  I did not have problems running ansible with the current contents of `group_vars/all`...

Comment (djmitche) [inline localhost]: Yep, see the description of http://trac.buildbot.net/ticket/3132

Comment (djmitche) [inline localhost]: Ah, it does work without this -- you just have to run ansible-pull in the repository directory.

**Comment by djmitche** [comment]
I'd like to get this merged ASAP, as things seem to keep breaking with Ansible and it'd be nice to have lots of cronspam when that occurs.

**Comment by djmitche** [comment]
Hm, Ansible stands falsely accused.  I think my method of trying to run ansible-pull by hand was incorrect.

**Comment by djmitche** [comment]
Deployed to:
 * [ ] service1
 * [ ] service2
 * [ ] service3
 * [ ] vm1
 * [ ] buildbot
 * [ ] docs
 * [ ] nine
 * [ ] ns1
 * [ ] www

**Pull request merged**